src/MessagePack.GeneratorCore/Generator/StringKey/StringKeyFormatterTemplate.tt

```diff
@@ -0,0 +1,284 @@
+﻿<#@ template debug="false" hostspecific="false" linePragmas="false" language="C#" #>
+<#@ assembly name="System.Core" #>
+<#@ import namespace="System" #>
+<#@ import namespace="System.Linq" #>
+<#@ import namespace="MessagePackCompiler.CodeAnalysis" #>
+// <auto-generated>
+// THIS (.cs) FILE IS GENERATED BY MPC(MessagePack-CSharp). DO NOT CHANGE IT.
+// </auto-generated>
+
+#pragma warning disable 618
+#pragma warning disable 612
+#pragma warning disable 414
+#pragma warning disable 168
+
+#pragma warning disable SA1129 // Do not use default value type constructor
+#pragma warning disable SA1200 // Using directives should be placed correctly
+#pragma warning disable SA1309 // Field names should not begin with underscore
+#pragma warning disable SA1312 // Variable names should begin with lower-case letter
+#pragma warning disable SA1403 // File may only contain a single namespace
+#pragma warning disable SA1649 // File name should match first type name
+
+namespace <#= Namespace #>
+{
+    using System;
+    using System.Buffers;
+    using System.Runtime.InteropServices;
+    using MessagePack;
+
+<#
+foreach(var objInfo in ObjectSerializationInfos)
+{
+    string formatterName = objInfo.Name + (objInfo.TemplateParametersString == null ? "Formatter" :  ("Formatter"+ objInfo.TemplateParametersString));
+    bool isFormatterResolverNecessary = ShouldUseFormatterResolverHelper.ShouldUseFormatterResolver(objInfo.Members);
+#>
+    public sealed class <#= formatterName #> : global::MessagePack.Formatters.IMessagePackFormatter<<#= objInfo.FullName #>>
+    {
+<#
+    for(int i = 0; i < objInfo.Members.Length; i++)
+    {
+        var member = objInfo.Members[i];#>
+        private static byte[] ____stringByteKeys_<#= member.Name #>;
+<#
+    }#>
+
+        public <#= formatterName #>()
```

Comment:
You are using constructor to initialize static fields. Either use static constructor or move static this code into field initialization.